linux_os/guide/system/accounts/accounts-pam/set_password_hashing_algorithm/set_password_hashing_algorithm_logindefs/oval/shared.xml

```diff
@@ -1,42 +1,45 @@
 <def-group>
-  <definition class="compliance" id="set_password_hashing_algorithm_logindefs" version="2">
+  <definition class="compliance" id="{{{ rule_id }}}" version="2">
     {{{ oval_metadata("The password hashing algorithm should be set correctly in /etc/login.defs.") }}}
     <criteria operator="AND">
-      <criterion test_ref="test_etc_login_defs_encrypt_method" />
+      <criterion test_ref="test_set_password_hashing_algorithm_logindefs"/>
     </criteria>
   </definition>
 
-  <ind:variable_test id="test_etc_login_defs_encrypt_method" check="all" comment="The value of ENCRYPT_METHOD should be set appropriately in /etc/login.defs" version="1">
-    <ind:object object_ref="object_last_encrypt_method_instance_value" />
-    <ind:state state_ref="state_last_encrypt_method_instance_value" />
+  <ind:variable_test id="test_set_password_hashing_algorithm_logindefs" version="1"
+    check="all" comment="The value of ENCRYPT_METHOD should be set appropriately in /etc/login.defs">
+    <ind:object object_ref="object_set_password_hashing_algorithm_logindefs"/>
+    <ind:state state_ref="state_set_password_hashing_algorithm_logindefs"/>
   </ind:variable_test>
 
   <ind:textfilecontent54_object id="object_last_encrypt_method_from_etc_login_defs" version="1">
     <!-- Read whole /etc/login.defs as single line so we can retrieve last ENCRYPT_METHOD directive occurrence -->
-    <ind:behaviors singleline="true" />
+    <ind:behaviors singleline="true"/>
     <ind:filepath>/etc/login.defs</ind:filepath>
     <!-- Retrieve last (uncommented) occurrence of ENCRYPT_METHOD directive -->
     <ind:pattern operation="pattern match">.*\n[^#]*(ENCRYPT_METHOD\s+\w+)\s*\n</ind:pattern>
     <ind:instance datatype="int" operation="greater than or equal">1</ind:instance>
   </ind:textfilecontent54_object>
 
   <!-- Capture the actual ENCRYPT_METHOD string value from the previously retrieved last instance -->
-  <local_variable id="variable_last_encrypt_method_instance_value" datatype="string" comment="The value of last ENCRYPT_METHOD directive in /etc/login.defs" version="1">
+  <local_variable id="variable_last_encrypt_method_instance_value" version="1"
+    datatype="string" comment="The value of last ENCRYPT_METHOD directive in /etc/login.defs">
     <regex_capture pattern="ENCRYPT_METHOD\s+(\w+)">
-      <object_component item_field="subexpression" object_ref="object_last_encrypt_method_from_etc_login_defs" />
+      <object_component item_field="subexpression" object_ref="object_last_encrypt_method_from_etc_login_defs"/>
```

Comment:
The frequent style updates make it difficult for the PR reviewers to focus on substance.

Reply:
I know, but is the first thing I usually to before any change in the logic. It is unfortunate we still have a lot of OVAL very difficult to be read because of lack of styles. For me it is much faster to first make it decently readable and only then improve the logic. OVAL already makes everything more boring and complex by nature. We can at least try to make it more readable.